### PR TITLE
codeintel: Handle ConfigurationPolicy nil case

### PR DESCRIPTION
**Pull request by eseliger**
The ConfigurationPolicy can be nil, but this logic didn't account for it, causing a panic.

## Test plan

Verified it no longer panics.

### CODE REVIEW

**Review by Numbers88s — APPROVED** [review]
Thanks 🙏

File: enterprise/cmd/frontend/internal/codeintel/resolvers/graphql/util.go [hunk 1]
```diff
@@ -257,6 +257,7 @@ func sharedRetentionPolicyToStoreRetentionPolicy(policy []policiesShared.Retenti
 				IndexIntermediateCommits:  p.IndexIntermediateCommits,
 			}
 		}
+
```
Comment (efritz) [inline enterprise/cmd/frontend/internal/codeintel/resolvers/graphql/util.go:260]: all I saw at first was this change XD

_that title doesn't match_

**Review by efritz — APPROVED** [review]

**Comment by eseliger** [comment]
`git commit --allow-empty -m "Trigger build"` might also work btw @Strum355

**Comment by Strum355** [comment]
> `git commit --allow-empty -m "Trigger build"` might also work btw @Strum355

Probably, was more convenient to do it just in github UI :p

**Pull request merged**